mod/01_core/format/html_format.rb

```diff
@@ -216,8 +265,8 @@ def unsupported_view view
     end
 
     def final_link href, opts={}
-      text = opts[:text] || href
-      %{<a class="#{opts[:class]}" href="#{href}">#{text}</a>}
+      text = (opts.delete(:text) || href).dup
+      content_tag :a, text, opts.merge(:href=>href)
```

Comment:
sorry, I followed that stackoverflow link, and I think that pretty well settles the last issue I had.  It looks like you addressed all my other concerns, so we're good now; I'll get this merged into 1.14.5.

I should have said this before, btw, but I really love that you consolidated these two.  I'm not yet totally in love with our link handling, but this was a BIG step in the right direction!!